### PR TITLE
Use starknet-rs types in rpc

### PR DESCRIPTION
**Pull request by ClementWalter**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Feature
- Refactoring (no functional changes, no API changes)

## What is the current behavior?

The RPC crates define custom types while:
- we strive to have one single typing for the whole starknet X rust ecosystem
- it takes useless time to write them, especially when they are complex
- it's bug prone

Resolves: #359 

## What is the new behavior?

Use only types from starknet-rs.

## Does this introduce a breaking change?

Yes and No.
We need to do #381 to use these types in the runtime as well. In the current state, H256 cannot be converted to FieldElements, not sure about what's going on.

### CODE REVIEW

File: crates/client/rpc-core/Cargo.toml [hunk 1]
```diff
@@ -28,3 +28,5 @@ sp-blockchain = { workspace = true, default-features = true }
 starknet_api = { workspace = true, default-features = false }
 frame-support = { workspace = true }
 sp-core = { workspace = true }
+starknet-providers = { git = "https://github.com/xJonathanLEI/starknet-rs", branch = "dev/jsonrpc_0_3_0", default-features = false }
```
Comment (abdelhamidbakhta) [inline crates/client/rpc-core/Cargo.toml:31]: would be better to use `workspace = true` here to inherit the version defined in the workspace Cargo toml.

**Review by abdelhamidbakhta — DISMISSED** [review]
left one very minor comment, otherwise looks good to me

**Comment by ClementWalter** [comment]
Need to do https://github.com/keep-starknet-strange/madara/issues/381 ASAP to remove all the H256>FieldElement conversions

**Review by EvolveArt — APPROVED** [review]
lgtm

**Pull request merged**